### PR TITLE
chore: improve rich text editor performance

### PR DESCRIPTION
**Pull request by pregnantboy**
## Problem
Rich text editors gets unresponsive and laggy when user create a lot of rows in MultiRow fields.

This is caused by the re-rendering of the entire react-hook-form onChange. 

## Solution

The re-rendering of the entire form is due to a useWatch hook at the form level that listens to changes to dynamically render conditional fields (hiddenIf). The hook has been removed in favour of a watch function that only detects changes in related fields. (for example, if field A depends on field B, it will only re-render if field B changes)

## What to test
- [ ] Check that hiddenIf fields still work
- [ ] Check that required fields work
- [ ] Check that form fields are saved properly.
- [ ] Lastly, check that creating multiple rows in MultiRow fields do not result in performance degradation

### CODE REVIEW

**Comment by pregnantboy** [comment]
* **#739** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/739?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop-v2`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @pregnantboy and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by liangyuanruo — APPROVED** [review]
lgtm

File: packages/frontend/src/graphql/client.ts [hunk 1]
```diff
@@ -13,6 +15,7 @@ const GRAPHQL_URL = '/graphql'
 const client = new ApolloClient({
   cache,
   link: createLink({ uri: GRAPHQL_URL }),
+  connectToDevTools: appConfig.isDev,
```
Comment (m0nggh) [inline packages/frontend/src/graphql/client.ts:18]: can i check whats this for ah

File: packages/frontend/src/components/RichTextEditor/Suggestions.tsx [hunk 2]
```diff
@@ -98,3 +98,5 @@ export default function Suggestions(props: SuggestionsProps) {
     </Flex>
   )
 }
+
+export default memo(Suggestions)
```
Comment (m0nggh) [inline packages/frontend/src/components/RichTextEditor/Suggestions.tsx:102]: actly what rendering does this reduce ah

**Review by m0nggh — APPROVED** [review]
tested that required and optional fields on old pipes still work, fields with hiddenif works

**Pull request merged**